File: 2624-difference-between-element-sum-and-digit-sum-of-an-array/difference-between-element-sum-and-digit-sum-of-an-array.kt
	class Solution {
		fun differenceOfSum(nums: IntArray): Int {
			val elementsSum = nums.flatMap { it.convertUnderTen() }.sum()
			val digitsSum = nums.sum()
			return digitsSum - elementsSum
		}
		private fun  Int.convertUnderTen(): List<Int> {
			return if ( this < 10)
				listOf(this)
			else
				this.toString().map { it.toString().toInt() }
		}
	}

File: 2624-difference-between-element-sum-and-digit-sum-of-an-array/difference-between-element-sum-and-digit-sum-of-an-array.swift
class Solution {
    func differenceOfSum(_ nums: [Int]) -> Int {
        let elementSum = nums.reduce(0) { $0 + $1 }
        let digitSum = nums.map { String($0) }
                            .flatMap { $0.map { Int(String($0))! } }
                            .reduce(0) { $0 + $1 }
        return abs(elementSum - digitSum)
    }
}